test/unittests/static/idle-timeout_spec.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/*
+ * Unit tests for the idle-timeout functionality.
+ */
+
+const requirejs = require("requirejs");
+
+// module mocks use relative path to the baseUrl
+const hereMocks = '../../../../test/unittests/static/';
+
+requirejs.config({
+  baseUrl: '../sources/web/datalab/static/',
+  nodeRequire: require,
+  map: {
+    '*': {
+      'base/js/dialog': hereMocks + 'dialog-mock',
+      'base/js/events': hereMocks + 'events-mock',
+    },
+  },
+});
+
+const idleTimeout = requirejs('idle-timeout');
+
+describe('Unit tests', function() {
+describe('Idle-timeout', function() {
+
+  describe('secondsToString', function() {
+    it('converts zero and negative to empy string', function() {
+      expect(idleTimeout._secondsToString(0)).toEqual('');
+      expect(idleTimeout._secondsToString(-1)).toEqual('');
+    });
+    it('converts single-unit strings correctly', function() {
+      expect(idleTimeout._secondsToString(1)).toEqual('1s');
+      expect(idleTimeout._secondsToString(15)).toEqual('15s');
+      expect(idleTimeout._secondsToString(2*60)).toEqual('2m');
```

Comment:
Multi-unit outputs are in the next test, a few lines lower. Is there something you want me to add there?